lib/utility/es_client.rb

```diff
@@ -57,8 +57,23 @@ def bulk(arguments = {})
 
     def raise_if_necessary(response)
       if response['errors']
-        first_error = response['items'][0]
-        raise IndexingFailedError.new("Failed to index documents into Elasticsearch.\nFirst error in response is: #{first_error.to_json}")
+        first_error = nil
+
+        response['items'].each do |item|
+          ['index', 'delete'].each do |op|
+            if item.has_key?(op) && item[op].has_key?('error')
+              first_error = item
+
+              break
+            end
+          end
+        end
+
+        if first_error
```

Comment:
that stops the whole sync. shouldn't we ignore just the doc and report back?

Reply:
I think for now we're okay - ignoring it will cause minor problems with document counters + we don't report these errors well to UI, so it might be confusing for now for the customer.

In future work (resilience) we want to make it better